Software/Development/SemesterProject/App/closedLoopControlManager.c

```diff
@@ -158,6 +155,67 @@ dqObject calculateVoltageReferences(dqObject currentReferences, dqObject dqCurre
     return dqVoltages;
 }
 
+/*
+ * This function performs a controlled increase of the control parameters
+ * in order to have a smooth startup. The parameters are:
+ *  +   PI_Ratio
+ *  +   MAX_DUTY_CYCLE
+ *  The increase is proportional to the rotor speed: when the speed is 0
+ *  the parameters will provide the most dampened response.
+ */
+void handleControlParameters(void)
+{
+    int16 rotorSpeed = abs(readRotorRPM());
 
+    maxDuty = getDutyRatio(rotorSpeed) * MAX_DUTY_CYCLE;
+    updatePIRatio(getPIRatio(rotorSpeed));
+}
+
+
+float getDutyRatio(int16 rotorSpeed)
+{
+    float dutyRatio = 0;
+    dutyRatio = (float)(rotorSpeed - SPEED_DUTY_MIN_DUTY) * FITTING_FUNCTION_SLOPE_DUTY + MIN_DUTY;
+
+    if (dutyRatio > MAX_DUTY)   dutyRatio = MAX_DUTY;
+    if (dutyRatio < MIN_DUTY) dutyRatio = MIN_DUTY;
+
+    return dutyRatio;
+}
+
+float getPIRatio(int16 rotorSpeed)
+{
+    float PIRatio = 0;
+    PIRatio = (float)(rotorSpeed - SPEED_DUTY_MIN_PI) * FITTING_FUNCTION_SLOPE_PI + MIN_DUTY;
+    if (PIRatio > MAX_DUTY)   PIRatio = MAX_DUTY;
+    if (PIRatio < MIN_DUTY) PIRatio = MIN_DUTY;
 

```

Comment:
I know we talked about this already, but i think it makes sense to have a different MIN_PI and MAX_PI parameters, otherwise this is a bit confusing...